opm/simulators/wells/WellFilterCake.cpp

```diff
@@ -37,106 +37,171 @@ namespace Opm {
 
 template<class Scalar>
 void WellFilterCake<Scalar>::
-updateFiltrationParticleVolume(const WellInterfaceGeneric<Scalar>& well,
-                               const double dt,
-                               const Scalar conc,
-                               const std::size_t water_index,
-                               WellState<Scalar>& well_state)
+updatePostStep(const WellInterfaceGeneric<Scalar>& well,
+               WellState<Scalar>& well_state,
+               const double dt,
+               const Scalar conc,
+               const std::size_t water_index,
+               DeferredLogger& deferred_logger)
 {
-    if (!well.isInjector()) {
+    if (! well.isInjector() || well.wellEcl().injectorType() != InjectorType::WATER)
         return;
-    }
-
-    if (filtration_particle_volume_.empty()) {
-        const auto& ws = well_state.well(well.indexOfWell());
-        filtration_particle_volume_.assign(ws.perf_data.size(), 0.); // initializing to be zero
-    }
 
-    const auto injectorType = well.wellEcl().injectorType();
-    if (injectorType != InjectorType::WATER) {
-        return;
+    assert (conc > 0.);
+    auto& ws = well_state.well(well.indexOfWell());
+    const auto nperf = ws.perf_data.size();
+    if (prev_skin_factor_.empty()) {
+        prev_skin_factor_.assign(nperf, 0.);
+        skin_factor_.assign(nperf, 0.);
+        thickness_.assign(nperf, 0.);
+        prev_thickness_.assign(nperf, 0.);
     }
 
-    auto& ws = well_state.well(well.indexOfWell());
+    prev_skin_factor_ = skin_factor_;
+    prev_thickness_ = thickness_;
```

Comment:
Maybe I misunderstand, but I think they need to be before the method exactly because they are updated in that function.

Reply:
I understand your point, with current way, the prev_thickness_ will be the thickness from the last time step.  Then in the coming time step, when we do cleaning, we should probably use `thickness_` instead of `prev_thickness_`. 

I think part of the confusion is that, we might not need to store `prev_thickness_` and `prev_skin_factor_` at all. They can be temporary variables inside the function `updateMultipliers()` (if indeed `thickness_` should be used in applyCleaning()).

Reply:
Ah, you're right - thanks! Will update.